pkg/gds/store/trtl/trtl.go

```diff
@@ -154,8 +277,23 @@ func (s *Store) UpdateVASP(v *gds.VASP) (err error) {
 		return err
 	}
 
-	// TODO: Update the indices.
+	// Critical section (optimizing for safety rather than speed)
+	s.Lock()
+	defer s.Unlock()
+
+	// Retrieve the original record to ensure that the indices are updated properly
+	// This must be inside the lock so that the database indices are consistent.
+	// NOTE: the lock doesn't prevent concurrent writes from multiple GDS instances,
+	// just from this instance ... we need trtl transactions to guarantee this (or
+	// indices managed by trtl itself, since this is the only reason we're doing a Get).
+	o, err := s.RetrieveVASP(v.Id)
+	if err != nil {
+		return err
+	}
 
+	// Update the VASP record
+	// This must be inside the lock so that there is no race condition between the index
+	// and the stored index inside of the database.
 	ctx, cancel := withContext(context.Background())
```

Comment:
The use of context here got me thinking that in the future we might want to revisit how we're using contexts in gds/trtl so that we can cancel long-running transactions that have locked the database

Reply:
SC-3652